### PR TITLE
ci(spanner): improve performance of samples tests

### PR DESCRIPTION
**Pull request by sakthivelmanii**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-spanner/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: samples/snippets/src/test/java/com/example/spanner/SpannerSampleIT.java [hunk 1]
```diff
@@ -51,12 +52,14 @@
...
 import org.junit.BeforeClass;
 import org.junit.Test;
+import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
 /** Unit tests for {@code SpannerSample} */
 @RunWith(JUnit4.class)
 @SuppressWarnings("checkstyle:abbreviationaswordinname")
+@Category(SlowTest.class)
```
Comment (sakthivelmanii) [inline samples/snippets/src/test/java/com/example/spanner/SpannerSampleIT.java:62]: These two tests were taking almost an hour. Plan is to introduce a job for slow sample integration tests(Java 11) to speed up the performance.

Comment (rahul2393) [reply samples/snippets/src/test/java/com/example/spanner/SpannerSampleIT.java:62]: Should we mark all the tests in SpannerSample as slow, or particualr backup/restore should be tagged?

Comment (sakthivelmanii) [reply samples/snippets/src/test/java/com/example/spanner/SpannerSampleIT.java]: For now, I have marked both the files to not to run given that it's taking an almost an hour. and we are running these tests for each module. Goal is to run only once either with [install-without-bom] or [snapshot] or [snippets]

File: samples/install-without-bom/pom.xml [hunk 2]
```diff
@@ -147,6 +147,8 @@
         <artifactId>maven-failsafe-plugin</artifactId>
         <version>3.5.2</version>
         <configuration>
+          <forkCount>10</forkCount>
+          <reuseForks>false</reuseForks>
```
Comment (olavloite) [inline samples/install-without-bom/pom.xml:151]: I don't think this does anything at the moment, as there are no integration tests defined in this project. Are we planning on adding tests here?

File: samples/snapshot/pom.xml [hunk 3]
```diff
@@ -146,6 +146,8 @@
         <artifactId>maven-failsafe-plugin</artifactId>
         <version>3.5.2</version>
         <configuration>
+          <forkCount>10</forkCount>
+          <reuseForks>false</reuseForks>
```
Comment (olavloite) [inline samples/snapshot/pom.xml:150]: Same as for `install-without-bom`: There are no tests in this project at the moment, so I don't think this does anything

File: samples/snippets/pom.xml [hunk 4]
```diff
@@ -187,6 +189,9 @@
             <spanner.sample.database>mysample</spanner.sample.database>
             <spanner.quickstart.database>quick-db</spanner.quickstart.database>
           </systemPropertyVariables>
+          <excludes>
+            <exclude>**/SpannerSampleIT.java</exclude>
+          </excludes>
```
Comment (olavloite) [inline samples/snippets/pom.xml:194]: I understand that this one is slow, but do we really want to skip it all-together? And/or will we have some other nightly job that overrides this configuration and does include this test?

File: samples/install-without-bom/pom.xml (hunk 2 above)
Comment (sakthivelmanii) [reply samples/install-without-bom/pom.xml:151]: It is. Currently we are runnings tests three times in all three modules.

File: samples/snapshot/pom.xml (hunk 3 above)
Comment (sakthivelmanii) [reply samples/snapshot/pom.xml:150]: It is. Currently we are runnings tests three times in all three modules.

File: samples/snippets/pom.xml (hunk 4 above)
Comment (sakthivelmanii) [reply samples/snippets/pom.xml:194]: I am introducing a new job to run this test alone in parallel so that it can be completed in an hour.

**Review by olavloite — APPROVED** [review]

**Pull request merged**